src/provision-dialog.ts

```diff
@@ -240,6 +239,10 @@ class ProvisionDialog extends LitElement {
   protected updated(changedProps: PropertyValues) {
     super.updated(changedProps);
 
+    if (changedProps.has("_state")) {
+      this.stateUpdateCallback(this._state);
```

Comment:
This is not the full "state". If `this._state === "improv-state"`, we should instead look at `this._improvCurrentState` for the current state.